src/node_report.cc

```diff
@@ -795,52 +900,54 @@ void PrintNativeStack(FILE* fp) {
  * Function to print a native stack backtrace - AIX
  *
  ******************************************************************************/
-void PrintNativeStack(FILE* fp) {
-  fprintf(fp, "\n================================================================================");
-  fprintf(fp, "\n==== Native Stack Trace ========================================================\n\n");
-  fprintf(fp, "Native stack trace not supported on AIX\n");
+void PrintNativeStack(std::ostream& out) {
+  out << "\n================================================================================";
+  out << "\n==== Native Stack Trace ========================================================\n\n";
+  out << "Native stack trace not supported on AIX\n";
 }
 #else
 /*******************************************************************************
  * Function to print a native stack backtrace - Linux/OSX
  *
  ******************************************************************************/
-void PrintNativeStack(FILE* fp) {
+void PrintNativeStack(std::ostream& out) {
   void* frames[256];
-  fprintf(fp, "\n================================================================================");
-  fprintf(fp, "\n==== Native Stack Trace ========================================================\n\n");
+  char buf[64];
```

Comment:
On this path the buffer only ever has to hold the result of `"%2d: [pc=0x%p]"` which should fit into 64 characters.